backend/repository/config.go

```diff
@@ -65,12 +65,43 @@ type EndpointConfig struct {
 	WorkflowType     string                    `json:"workflowType"`
 	ClientID         string                    `json:"clientID"`
 	ClientMethod     string                    `json:"clientMethod"`
-	TestFixtures     []string                  `json:"testFixtures"`
+	TestFixtures     []*TestStub                  `json:"testFixtures"`
 	Middlewares      []*codegen.MiddlewareSpec `json:"middlewares"`
 	ReqHeaderMap     map[string]string         `json:"reqHeaderMap"`
 	ResHeaderMap     map[string]string         `json:"resHeaderMap"`
 }
 
+// TestStub saves stubbed requests/responses for an endpoint test.
+type TestStub struct {
+	TestName               string `json:"testName"`
+	EndpointID             string `json:"endpointId"`
+	HandlerID              string `json:"handlerId"`
+	EndpointRequest        map[string]interface{} `json:"endpointRequest"`
+	EndpointRequestString  string `json:"endpointRequestString"`
+	EndpointReqHeaders     map[string]string      `json:"endpointReqHeaders"`
+	EndpointReqHeaderKeys  []string               `json:"endpointReqHeaderKeys"`
```

Comment:
Why headerKeys are needed as you already have header map?

Reply:
same above